src/js/control-bar/text-track-controls/text-track-menu-item.js

```diff
@@ -93,27 +93,36 @@ class TextTrackMenuItem extends MenuItem {
    * @listens click
    */
   handleClick(event) {
-    const kind = this.track.kind;
-    let kinds = this.track.kinds;
+    const referenceTrack = this.track;
     const tracks = this.player_.textTracks();
 
-    if (!kinds) {
-      kinds = [kind];
-    }
-
     super.handleClick(event);
 
     if (!tracks) {
       return;
     }
 
+    // Determine the relevant kind(s) of tracks for this component.
+    const kinds = (referenceTrack.kinds || [referenceTrack.kind]).filter(Boolean);
```

Comment:
After looking at the tests I was going to suggest that we add a tests for a track with multiple kinds. Then I looked at the HTML5 video spec and our track implementation and it seems like `kinds` isn't actually a thing. Only `kind` should ever exist. Perhaps we can keep it for now but add some code about deprecating this usage?

Reply:
`kinds` is a thing that was added for the combined subtitles/captions menu, it should be considered an implementation detail and not used outside of Video.js internals.

Reply:
Yeah, it might have been good to name it in a way that was more obviously an implementation detail because that confused me at first, too. Oh well.